src/test/java/com/google/sps/youtube/YoutubeInfoScraperTest.java

```diff
@@ -0,0 +1,96 @@
+import java.util.Arrays;
+import com.google.api.services.youtube.model.Channel;
+import com.google.api.services.youtube.model.ChannelContentDetails.RelatedPlaylists;
+import com.google.api.services.youtube.model.ChannelContentDetails;
+import com.google.api.services.youtube.model.ChannelListResponse;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.youtube.YouTubeRequestInitializer;
+import com.google.api.services.youtube.YouTube;
+import java.io.IOException;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public final class YoutubeInfoScraperTest {
+
+  private YoutubeInfoScraper scrape;
+  private ChannelListResponse mockResponse;
+  private YoutubeResponse apiResponse; 
+    
+  @Before
+  public void setUp() {
+    apiResponse = mock(YoutubeResponse.class);
+    scrape = scrape.setYoutubeAndGetNewInstance(
+        new YouTube.Builder(new NetHttpTransport(),
+        JacksonFactory.getDefaultInstance(),
+        null)
+        .setApplicationName("APPLICATION_NAME")
+        .setYouTubeRequestInitializer(new YouTubeRequestInitializer("API_KEY"))
+        .build());
+  }
+
+  @Test
+  public void scrapeChannelUploadPlaylist_nonExistentChannelId() throws Exception{
+      ChannelListResponse mockResponse = new ChannelListResponse();
+      when(apiResponse.getYoutubeChannelListResponse("CHANNELID_THAT_DOES_NOT_EXIST")).thenReturn(mockResponse.setItems(null));
+      String actual = scrape.scrapeChannelUploadPlaylist(apiResponse, "CHANNELID_THAT_DOES_NOT_EXIST");
+      Assert.assertEquals("null", actual);
+  }
+
+ @Test
+  public void scrapeChannelUploadPlaylist_emptyList() throws Exception{
+      ChannelListResponse mockResponse = new ChannelListResponse();
+      when(apiResponse.getYoutubeChannelListResponse("CHANNELID")).thenReturn(mockResponse.setItems(Arrays.asList()));
+      String actual = scrape.scrapeChannelUploadPlaylist(apiResponse, "CHANNELID");
+      Assert.assertEquals("null", actual);
+  }
+
+  @Test
+  public void scrapeChannelUploadPlaylist_channelExists() throws Exception{
+      mockResponse = new ChannelListResponse();
+      Channel channel = new Channel();
+      channel.setContentDetails(new ChannelContentDetails().setRelatedPlaylists(
+          new RelatedPlaylists().setUploads("UPLOAD_ID")));
+      mockResponse.setItems(Arrays.asList(channel));
+      when(apiResponse.getYoutubeChannelListResponse("CHANNELID_THAT_EXISTS")).thenReturn(mockResponse);
+      String actual = scrape.scrapeChannelUploadPlaylist(apiResponse, "CHANNELID_THAT_EXISTS");
+      Assert.assertEquals("UPLOAD_ID", actual);
+  }
+
+  @Test (expected = IOException.class)
```

Comment:
I would say go see: go/java-testing/exceptions 
:'(

but you can't so I'll just paste it below.

TL;DR is Use assertThrows, and all other ways to verify exceptions are discouraged.

---
Use assertThrows
The strongly preferred solution is to use Assert.assertThrows():

import static com.google.common.truth.Truth.assertThat;
import static org.junit.Assert.assertThrows;

import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

@RunWith(JUnit4.class)
public class AppendOnlyListTest {
  @Test
  public void removeFails() {
    AppendOnlyList list = new AppendOnlyList();
    list.add(0, "a");

    assertThrows(
        UnsupportedOperationException.class,
        () -> list.remove(0));
  }
}
The body of the lambda expression should always be a single statement, not a block.

Testing further details about the thrown exception
In some situations you will decide it is important to check additional details about the thrown exception. You might care about its message, or the attributes of its chained exception, for example.

It is easy to check these details, as assertThrows returns the thrown exception to you, and you can pass it to Truth:

  UnsupportedOperationException thrown =
      assertThrows(
          UnsupportedOperationException.class,
          () -> list.remove(0));
  assertThat(thrown).hasCauseThat().isInstanceOf(SomeOtherException.class);
  assertThat(thrown).hasMessageThat().contains("blah blah blah");
This test will fail if no exception is thrown, if the wrong type is thrown, or if either of the additional assertions on the exception fail.

Supplying additional message text
If you have any useful additional data to provide that will help the engineer debugging a failure, provide a custom message to assertThrows.

  private void assertBad(String input) {
    assertThrows(
        "input was: " + input,
        IllegalArgumentException.class,
        () -> process(input));
  }
Supplying a constant string adds little value, since a failure here always means the same thing: the method just called wasn't expected to complete normally, but it did.

Using the legacy try idiom
There is one case where you might choose not to use assertThrows:

When you aren't able to use Java 8 constructs, particularly lambda expressions. You can still use assertThrows, but you may find it bulky enough to outweigh its advantages.
However, use caution when doing so (read the Warnings below).

Example
import static com.google.common.truth.Truth.assertThat;
import static org.junit.Assert.fail;

import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

@RunWith(JUnit4.class)
public class AppendOnlyListTest {
  @Test
  public void removeFails() {
    AppendOnlyList list = new AppendOnlyList();
    list.add(0, "a");

    try {
      list.remove(0);
      fail(); // supply a message IF there is useful data to include
    } catch (UnsupportedOperationException expected) {
    }
  }
}
If you choose to provide a custom message to fail, see the advice above.

Warnings
Never use this idiom to test for AssertionError; it will malfunction and vacuously pass. Note that when testing production code, it should be impossible to cause an AssertionError to happen anyway. But unfortunately, this is also the exception type that some test libraries started using. Even more unfortunately, JUnit 4's regular assertion failures started using it, which is what causes the malfunction described here. Stick to assertThrows if at all possible. (If not, you will need to return from the catch block and move the fail after that.)
Like our advice for assertThrows, you should include only one statement in the try block.
Be careful not to forget the fail(), or your test will vacuously pass. (We try to trigger a warning in Critique in this case, but it doesn't catch everything.)
Use a variable name beginning with expected, as shown, or you will get a lint warning.
Note that all of the above problems can be avoided by using assertThrows.

All other solutions are discouraged
Don't use ExpectedException, @Test(expected =), or ExceptionExpectations. You can find a lengthy explanation of the pitfalls of these approaches in this page's history.

Migrate tests using these legacy utilities onto assertThrows using go/exception-test-migrator.

More advice
If your method should throw FooException, test for FooException
Don't test for a more generic type like RuntimeException or Exception. A test that expects RuntimeException can falsely pass if it contains a bug that triggers a NullPointerException.

When appropriate, perform additional assertions on the exception
to verify that the message contains diagnostic information -- for example, the value that failed its precondition check
to test that the exception happened for one reason and not another -- for example, that an IllegalArgumentException was caused by a -1 parameter and not by a "" parameter
To access an exception's cause consider Throwables.getCauseAs()
In addition to Truth's assertThat(e).hasCauseThat().... assertions, you may want to directly retrieve an exceptions' expected cause. Avoid doing it this way, however:

MyExpectedCause cause = (MyExpectedCause) thrown.getCause();
The problem is that if the cause is not of type MyExpectedCause, the original exception and stack trace will vanish in favor of the new ClassCastException. Instead, prefer Throwables.getCauseAs(), which will always safely preserve the original exception:

MyExpectedCause cause = Throwables.getCauseAs(thrown, MyExpectedCause.class);
Consider FutureSubject when asserting on failed futures
Although you can use assertThrows(ExecutionException.class, someFuture::get) to assert that a future fails, FutureSubject may be easier to use. There is also LabsFutures.getFailureCauseAs() to get the cause of the failed future.

Reply:
Changed to Asserthrows.  Had to update junit version in maven.